npm/pkg/dataplane/dataplane.go

```diff
@@ -0,0 +1,118 @@
+func (dp *DataPlane) InitializeDataPlane() error {
+	return dp.initializeDataPlane()
+}
+
+// ResetDataPlane helps in cleaning up dataplane sets and policies programmed
+// by NPM, retunring a clean slate
+func (dp *DataPlane) ResetDataPlane() error {
+	return dp.resetDataPlane()
+}
+
+// CreateIPSet takes in a set object and updates local cache with this set
+func (dp *DataPlane) CreateIPSet(set *ipsets.IPSet) error {
+	return dp.ipsetMgr.CreateIPSet(set)
+}
+
+// DeleteSet checks for members and references of the given "set" type ipset
+// if not used then will delete it from cache
+func (dp *DataPlane) DeleteSet(name string) error {
+	return dp.ipsetMgr.DeleteSet(name)
+}
+
+// DeleteList sanity checks and deletes a list ipset
+func (dp *DataPlane) DeleteList(name string) error {
+	return dp.ipsetMgr.DeleteList(name)
+}
+
+// AddToSet takes in a list of IPset objects along with IP member
+// and then updates it local cache
+func (dp *DataPlane) AddToSet(setNames []*ipsets.IPSet, ip, podKey string) error {
+	return dp.ipsetMgr.AddToSet(setNames, ip, podKey)
+}
+
+// RemoveFromSet takes in list of setnames from which a given IP member should be
+// removed and will update the local cache
+func (dp *DataPlane) RemoveFromSet(setNames []string, ip, podKey string) error {
+	return dp.ipsetMgr.RemoveFromSet(setNames, ip, podKey)
+}
+
+// AddToList takes a list name and list of sets which are to be added as members
+// to given list
+func (dp *DataPlane) AddToList(listName string, setNames []string) error {
+	return dp.ipsetMgr.AddToList(listName, setNames)
+}
+
+// RemoveFromList takes a list name and list of sets which are to be removed as members
+// to given list
+func (dp *DataPlane) RemoveFromList(listName string, setNames []string) error {
+	return dp.ipsetMgr.RemoveFromList(listName, setNames)
+}
+
+// UpdatePod is to be called by pod_controller ONLY when a new pod is CREATED.
+// this function has two responsibilities in windows
+// 1. Will call into dataplane and updates endpoint references of this pod.
+// 2. Will check for existing applicable network policies and applies it on endpoint
+// In Linux, this function currently is a no-op
+func (dp *DataPlane) UpdatePod(pod *npm.NpmPod) error {
+	return nil
+}
+
+// ApplyDataPlane all the IPSet operations just update cache and update a dirty ipset structure,
+// they do not change apply changes into dataplane. This function needs to be called at the
+// end of IPSet operations of a given controller event, it will check for the dirty ipset list
+// and accordingly makes changes in dataplane. This function helps emulate a single call to
+// dataplane instead of multiple ipset operations calls ipset operations calls to dataplane
+func (dp *DataPlane) ApplyDataPlane() error {
+	return dp.ipsetMgr.ApplyIPSets(dp.networkID)
+}
+
+// AddPolicies takes in a translated NPMNetworkPolicy object and applies on dataplane
+func (dp *DataPlane) AddPolicies(policies *policies.NPMNetworkPolicy) error {
```

Comment:
Can we make a consistent naming convention for function names and args around policies (with respect to plural vs. singular)?
In this file we have:
- AddPolicies(policies *NPMNetworkPolicy)  [plural, plural]
- RemovePolicies(policyName string)    [plural, singular]
and in policymanager.go we have:
-  AddPolicies(policy *NPMNetworkPolicy)   [plural, singular]
- GetPolicy(name string)   [singular, singular]

Even though a *NPMNetworkPolicy object can have multiple ACLs, I'd prefer using singular name and arg because plural usually implies a slice/array of the base object (string or *NPMNetworkPolicy).

Reply:
That makes sense. Changed the functions to reflect singular CRUD operations.